### PR TITLE
fix `--source`

### LINKED ISSUE #535: `--source` is broken
The `--source` option doesn't work correctly. For example:
```python
def foo():
    pass
```

```
> pydocstyle test.py --source

test.py:1 at module level:
        D100: Missing docstring in public module

 1: d 2: e 3: f 4:   5: f 6: o 7: o        ...


test.py:1 in public function `foo`:
        D103: Missing docstring in public function

 1: d 2: e 3: f 4:   5: f 6: o 7: o        ...
```

The problem is that [`Error.lines` in `pydocstyle/violations.py`](https://github.com/PyCQA/pydocstyle/blob/837c0c2ce7481fb20ae9af115203348961734e8c/src/pydocstyle/violations.py#L61) expects [`Definition.source` defined in `pydocstyle/parser.py`](https://github.com/PyCQA/pydocstyle/blob/837c0c2ce7481fb20ae9af115203348961734e8c/src/pydocstyle/parser.py#L118) to be a list of strings, but it's actually a single string with newlines in it. So instead of iterating over lines, it ends up iterating over characters in that string.

### PR DESCRIPTION
**Pull request by RuRo**
This PR closes #535.

Please make sure to check for the following items:
- [ ] Add unit tests and integration tests where applicable.
- [x] Add a line to the release notes (docs/release_notes.rst) under "Current Development Version".

### CODE REVIEW

**Comment by RuRo** [comment]
@samj1912 any comments for this PR? Also, you mentioned that you could make a new release soon, any progress on that?

File: docs/release_notes.rst [hunk 1]
```diff
@@ -20,6 +20,10 @@ New Features
 
 * Enable full toml configuration and pyproject.toml (#534).
 
+Bug Fixes
```
Comment (samj1912) [inline docs/release_notes.rst:23]: Can you move this up to Current dev version?

Comment (RuRo) [reply docs/release_notes.rst]: Done

**Review by samj1912 — APPROVED** [review]

**Comment by RuRo** [comment]
Sorry to bother you, but would you mind also releasing this fix right away while we're at it? I kind of originally expected to get this merged into `6.1.0`.

**Pull request merged**